### PR TITLE
1378 : Compress .rdlog with gzip compression

### PR DESCRIPTION
**Pull request by variacode95**
Hi @gschueler @miguelantonio 

This is enhancement to Rundeck that compresses the .rdlog output file with gzip compression and stores it with .rdlog.gz

Please review.

### CODE REVIEW

File: rundeckapp/src/groovy/com/dtolabs/rundeck/app/internal/logging/FSStreamingLogReader.groovy [hunk 1]
```diff
@@ -45,10 +48,21 @@ class FSStreamingLogReader implements ReverseSeekingStreamingLogReader {
         this.rundeckLogFormat=rundeckLogFormat
     }
     private void detectLegacyLogFile(){
-        file.withReader('UTF-8') {reader->
+
+        if(file.getName().contains(LoggingService.LOG_FILE_FILETYPE)) {
```
Comment (gschueler) [inline rundeckapp/src/groovy/com/dtolabs/rundeck/app/internal/logging/FSStreamingLogReader.groovy]: I think this should check for `file.getName().endsWith('.'+LoggingService. LOG_FILE_FILETYPE_UNCOMPRESSED)`

File: rundeckapp/src/groovy/com/dtolabs/rundeck/app/internal/logging/FSStreamingLogReader.groovy [hunk 2]
```diff
@@ -99,9 +113,17 @@ class FSStreamingLogReader implements ReverseSeekingStreamingLogReader {
     }
 
     private LogEventIterator beginFromOffset(long offset) {
-        def raf = new FileInputStream(file)
-        raf.channel.position(offset)
-        def LogEventIterator iterator = detectedIterator(new FSFileLineIterator(raf, encoding))
+        def LogEventIterator iterator
+
+        if(file.getName().contains(LoggingService.LOG_FILE_FILETYPE)) {
```
Comment (gschueler) [inline rundeckapp/src/groovy/com/dtolabs/rundeck/app/internal/logging/FSStreamingLogReader.groovy]: same as above

**Comment by gschueler** [comment]
the travis build failed during the functional tests here: https://travis-ci.org/rundeck/rundeck/jobs/146303991#L7868

FYI: you can run all of the functional tests locally doing this procedure:

    ./gradlew clean build # build everything
    bash run-api-tests.sh

**Comment by variacode95** [comment]
@gschueler @miguelantonio As per your instruction, I found it's failing in `run-api-tests.sh`.  I got similar error as [EOFException occurred when processing request: [GET] /api/5/execution/7/output.xml](https://travis-ci.org/rundeck/rundeck/jobs/146303991#L7893), when I executed `run-api-tests.sh`, but when I make api call `/api/5/execution/7/output.xml` it worked fine. 

Could you please guide how could I reproduce it?

**Comment by gschueler** [comment]
hey i think there is a fundamental problem here: while a job is running the output is written to the rdlog file, and rundeck expects to be able to open and read the file before it is "closed".  So if it is writing to a ZIP stream, this is probably the problem, as rundeck is trying to open and read the file before it is complete.

I think the gzip compression will have to happen after the file is completely closed, or in parallel (e.g. a Tee type stream)

**Comment by gschueler** [comment]
closing due to needing a different approach: @miguelantonio  we can revisit if needed

**Pull request closed without merging**